config.lua

```diff
@@ -46,25 +43,53 @@ config('development', {
     measure_performance = true
 })
 
-config('production', {
+config({'production', 'staging'}, {
+    hostname = os.getenv('HOSTNAME'),
+    secondary_hostname = os.getenv('SECONDARY_HOSTNAME'),
+    primary_cert_name = os.getenv('PRIMARY_CERT_NAME'),
+    secondary_cert_name = os.getenv('SECONDARY_CERT_NAME'),
+
     use_daemon = 'on',
-    enable_https = true,
     port = os.getenv('PORT') or 80,
-    site_name = 'Snap Cloud',
-    hostname = os.getenv('HOSTNAME') or 'cloud.snap.berkeley.edu',
-    ssl_cert_file = os.getenv('SSL_CERT_FILE'),
-    ssl_cert_private_key = os.getenv('SSL_CERT_PRIVATE_KEY'),
-    second_hostname = os.getenv('SECOND_HOSTNAME') or 'snap-cloud.cs10.org',
-    ssl_second_cert_file = os.getenv('SSL_SECOND_CERT_FILE'),
-    ssl_second_cert_private_key = os.getenv('SSL_SECOND_CERT_PRIVATE_KEY'),
+    ssl_port = os.getenv('SSL_PORT') or 443,
     dns_resolver = '67.207.67.2 ipv6=off',
 
     secret = os.getenv('SESSION_SECRET_BASE'),
-    num_workers = 8,
     code_cache = 'on',
 
     log_directive = 'logs/error.log warn',
 
+    compiled_files = {'berkeley', 'cs10.org'},
+
     -- TODO: See if we can turn this on without a big hit
     measure_performance = false
 })
+
+config('production', {
+    site_name = 'Snap Cloud',
+    num_workers = 8
+})
+
+config('staging', {
+    site_name = 'staging | Snap Cloud',
+    -- the staging server is a low-cpu server.
+    num_workers = 2
+})
+
+-- Compile nginx include files.
+local Compiler = require("lapis.cmd.nginx.config").ConfigCompiler
```

Comment:
This doesn't work on the staging server, but does locally. I'm scrapping it for a simpler approach, in a new PR.